pkg/asset/machines/alibabacloud/machines.go

```diff
@@ -35,8 +35,12 @@ func Machines(clusterID string, config *types.InstallConfig, pool *types.Machine
 
 	var machines []machineapi.Machine
 	for idx := int64(0); idx < total; idx++ {
-		azIndex := int(idx) % len(azs)
-		provider, err := provider(clusterID, platform, mpool, azIndex, role, userDataSecret, resourceTags)
+		zoneID := azs[int(idx)%len(azs)]
+		vswitchID, ok := vswitchMaps[zoneID]
+		if len(vswitchMaps) > 0 && !ok {
+			return nil, errors.Errorf("no VSwitch for zone %s", zoneID)
```

Comment:
This check is fine to leave here, but the user should never encounter this error. This must be caught by earlier validation.

Reply:
You are right, I removed it